pkg/system/phase4_configuring.go

```diff
@@ -682,6 +688,97 @@ func (r *Reconciler) RegisterToCluster() error {
 	return r.NBClient.RegisterToCluster()
 }
 
+// ReconcileDefaultNamespaceStore checks if the default nsfs pvc exist or not
+// and attempts to create default nsfsnamespacestore using nsfs pvc which inturn uses
+// spectrum scale storage class.
+func (r *Reconciler) ReconcileDefaultNamespaceStore() error {
+	// Skip if joining another NooBaa
+	if r.JoinSecret != nil {
+		return nil
+	}
+
+	log := r.Logger.WithField("func", "ReconcileDefaultNamespaceStore")
+
+	if r.DefaultNsfsPvc.UID == "" {
+		log.Infof("PVC %s does not  exist. skipping Reconcile %s", r.DefaultNsfsPvc.Name, r.DefaultNamespaceStore.Name)
+		return nil
+	}
+
+	util.KubeCheck(r.DefaultNamespaceStore)
+	// Namespacestore already exist
+
+	if r.DefaultNamespaceStore.UID != "" {
+		log.Infof("NamespaceStore %s already exists. skipping Reconcile", r.DefaultNamespaceStore.Name)
+		return nil
+	}
+
+	// If Manual default backing store is enable, skip creating default namespacestore
+	if r.NooBaa.Spec.ManualDefaultBackingStore {
+		r.Logger.Info("ManualDefaultBackingStore is true, Skip Reconciling nsfs namespacestore")
+		return nil
+	}
+
+	r.DefaultNamespaceStore.Spec.Type = nbv1.NSStoreTypeNSFS
+	r.DefaultNamespaceStore.Spec.NSFS = &nbv1.NSFSSpec{}
+	r.DefaultNamespaceStore.Spec.NSFS.PvcName = r.DefaultNsfsPvc.Name
+	r.DefaultNamespaceStore.Spec.NSFS.SubPath = ""
+
+	r.Own(r.DefaultNamespaceStore)
+
+	if err := r.Client.Create(r.Ctx, r.DefaultNamespaceStore); err != nil {
+		log.Errorf("got error on DefaultNamespaceStore creation. error: %v", err)
+		return err
+	}
+	return nil
+}
+
+// ReconcileDefaultNsfsPvc checks if the noobaa is running on Fusion HCI with
+// spectrum scale and attempts to create default PVC for nsfs using spectrum scale
+// storage class.
+func (r *Reconciler) ReconcileDefaultNsfsPvc() error {
+	// Skip if joining another NooBaa
+	if r.JoinSecret != nil {
+		return nil
+	}
+
+	// Check if ODF is installed on Fusion HCI cluster with spectrum scale.
+	if !util.IsFusionHCIWithScale() {
+		r.Logger.Info(" IBM Fusion HCI with spectrum scale NOT detected.")
```

Comment:
You are right. We should not print this log for every system which is not HCI. 
However, I would like this message to be printed, mainly because we have not checked this code on HCI.
Now, I am logging it only when we detect HCI. That will help us in debugging also and at the same time it will not print it for every other platform.